packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -697,6 +699,18 @@ export abstract class QueryBuilderState implements CommandRegistrar {
         );
       }
       if (this.parametersState.parameterStates.length > 0) {
+        const fetchStructureStateImp = this.fetchStructureState.implementation;
```

Comment:
if this is required could we add a test for this case as to why its needed.

Reply:
when doing a roundtrip from text to form mode, MilestoningState is not initialized if there are only milestoned-derived-properties that aren't coming from the root class ( root class could be not milestoned).  (p.s. but there are still parameters. not just link to MilestoningState)

Since we are going to support default values for milestoning parameters, we should initialize MilestoningState.


Current existing  test cases already covers this case

Reply:
It's not a bug. 
There are still parameters named e.g.  processingDate.. so when running queries, setting param values modal will be open and users have to pass in values. 


Since default values are required for milestoning parameters and based on my current logic of implementing this, MilestoningState has to be initialized as the case where the root class is milestoning.


There are already several tests in `QUERY_PROPERTY_PARAMETER_CASES`